### PR TITLE
Fix panic on format nil *fmt.Stringer type value

### PR DESCRIPTION
**Pull request by zaneli**
## Summary
Currently, if the pointer to the struct that implements `fmt.Stringer` is nil, the bind method will panic.
This PR fixes that.

## Checklist
Delete items not relevant to your PR:
- [x] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
  - Is this what I should do in this PR?
- ~~[ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials~~

### CODE REVIEW

File: bind_test.go [hunk 1]
```diff
@@ -394,6 +394,56 @@ func TestBindNamedWithTernaryOperator(t *testing.T) {
...
+		},
+		{
+			query:    "SELECT * FROM t WHERE null_coll = ?",
+			param:    nil,
+			expected: "SELECT * FROM t WHERE null_coll = NULL",
+		},
+		{
+			query:    "SELECT * FROM t WHERE null_coll = ?",
+			param:    (*typeForIssue1200)(nil),
+			expected: "SELECT * FROM t WHERE null_coll = NULL",
```
Comment (zaneli) [inline bind_test.go:423]: Before this modification

```
testing.tRunner.func1.2({0x1611d80, 0xc00003c840})
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1548 +0x397
panic({0x1611d80?, 0xc00003c840?})
	/Users/zaneli/sdk/go1.21.1/src/runtime/panic.go:914 +0x21f
github.com/ClickHouse/clickhouse-go/v2.(*typeForIssue1200).String(0xc00008a540?)
	<autogenerated>:1 +0x25
github.com/ClickHouse/clickhouse-go/v2.format(0x1c78060?, 0xe0?, {0x1604de0?, 0x0?})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/bind.go:309 +0x5ef
github.com/ClickHouse/clickhouse-go/v2.bindPositional(0x0?, {0x16c3fb0, 0x23}, {0xc0000d3e30, 0x1, 0x0?})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/bind.go:144 +0x2d4
github.com/ClickHouse/clickhouse-go/v2.bind(0x17ef640?, {0x16c3fb0, 0x23}, {0xc000075e30?, 0x1, 0x1})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/bind.go:93 +0x235
github.com/ClickHouse/clickhouse-go/v2.TestIssue1200(0x0?)
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/bind_test.go:433 +0x295
testing.tRunner(0xc0000ea680, 0x17571e0)
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 1
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1648 +0x3ad
FAIL	github.com/ClickHouse/clickhouse-go/v2	0.294s
FAIL
```

**Comment by zaneli** [comment]
Sorry, I realized the better way to fix it.
Please let me try again at another PR.

File: bind.go [hunk 2]
```diff
@@ -306,6 +306,11 @@ func format(tz *time.Location, scale TimeUnit, v any) (string, error) {
 		}
 		return fmt.Sprintf("[%s]", val), nil
 	case fmt.Stringer:
+		if v := reflect.ValueOf(v); v.Kind() == reflect.Pointer &&
+			v.IsNil() &&
+			v.Type().Elem().Implements(reflect.TypeOf((*fmt.Stringer)(nil)).Elem()) {
+			return "NULL", nil
```
Comment (zaneli) [inline bind.go:312]: I made a similar check using `database/sql` as a reference.
This implementation is also copied `go-sql-driver/mysql`.
Should I write a source code comment mentioning the reference source?

* https://cs.opensource.google/go/go/+/6ba6e72e39ac5be92f604832640c3131ad3d0ade:src/database/sql/driver/types.go;l=217-235
* https://github.com/go-sql-driver/mysql/blob/097fe6e3ad83bbd7c84debe810aec4c4a533bcaa/statement.go#L199-L217

File: tests/issues/1200_pr_test.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			want:  nil,
+		},
+		{
+			name:  "fmt.Stringer implemented struct pointer value",
+			value: &Test1200NullStr{underlying: v},
+			want:  &v,
+		},
+		{
+			name:  "fmt.Stringer implemented struct typed-nil value",
+			value: (*Test1200NullStr)(nil),
```
Comment (zaneli) [inline tests/issues/1200_pr_test.go:54]: Currently, panic this case.

```
--- FAIL: Test1200 (0.03s)
    --- FAIL: Test1200/fmt.Stringer_implemented_struct_typed-nil_value (0.00s)
panic: value method github.com/ClickHouse/clickhouse-go/v2/tests/issues.Test1200NullStr.String called using nil *Test1200NullStr pointer [recovered]
	panic: value method github.com/ClickHouse/clickhouse-go/v2/tests/issues.Test1200NullStr.String called using nil *Test1200NullStr pointer

goroutine 188 [running]:
testing.tRunner.func1.2({0x10084d7a0, 0xc0001ae200})
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1545 +0x238
testing.tRunner.func1()
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1548 +0x397
panic({0x10084d7a0?, 0xc0001ae200?})
	/Users/zaneli/sdk/go1.21.1/src/runtime/panic.go:914 +0x21f
github.com/ClickHouse/clickhouse-go/v2/tests/issues.(*Test1200NullStr).String(0xc00008a958?)
	<autogenerated>:1 +0x25
github.com/ClickHouse/clickhouse-go/v2.format(0x1010c8680?, 0x60?, {0x100839d60?, 0x0?})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/bind.go:309 +0x62e
github.com/ClickHouse/clickhouse-go/v2.bindPositional(0x0?, {0x1009644b4, 0x32}, {0xc0000d6160, 0x2, 0x0?})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/bind.go:144 +0x2d4
github.com/ClickHouse/clickhouse-go/v2.bind(0x0?, {0x1009644b4, 0x32}, {0xc0000d6160?, 0x2, 0x2})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/bind.go:93 +0x235
github.com/ClickHouse/clickhouse-go/v2.bindQueryOrAppendParameters(0x0?, 0xc00008ad58?, {0x1009644b4?, 0x32?}, 0x0?, {0xc0000d6160?, 0x2?, 0x2?})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/query_parameters.go:59 +0x16e
github.com/ClickHouse/clickhouse-go/v2.(*connect).exec(0xc000512000, {0x100ac6c00, 0x1010c8680}, {0x1009644b4, 0x32}, {0xc0000d6160, 0x2, 0x2})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/conn_exec.go:30 +0x105
github.com/ClickHouse/clickhouse-go/v2.(*clickhouse).Exec(0x5ad?, {0x100ac6c00, 0x1010c8680}, {0x1009644b4, 0x32}, {0xc0000d6160, 0x2, 0x2})
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/clickhouse.go:149 +0x99
github.com/ClickHouse/clickhouse-go/v2/tests/issues.Test1200.func2(0x1010c8680?)
	/Users/zaneli/go/src/github.com/zaneli/clickhouse-go/tests/issues/1200_pr_test.go:61 +0x126
testing.tRunner(0xc000622680, 0xc0003ee4c0)
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 185
	/Users/zaneli/sdk/go1.21.1/src/testing/testing.go:1648 +0x3ad
```

**Review by jkaflik — APPROVED** [review]
